tests/unit/MigrationTest.ts

```diff
@@ -0,0 +1,343 @@
+                            reportID: '4',
+                        },
+                        2: {
+                            reportActionID: '2',
+                            previousReportActionID: '23',
+                            created: '',
+                            actionName: CONST.REPORT.ACTIONS.TYPE.MARKEDREIMBURSED,
+                            reportID: '4',
+                        },
+                    },
+                ],
+                (item) => item[1].reportID ?? '',
+            );
+
+            return Onyx.multiSet({
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]: {},
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]: null,
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]: null,
+                ...reportActionsCollectionDataSet,
+            })
+                .then(CheckForPreviousReportActionID)
+                .then(() => {
+                    expect(LogSpy).toHaveBeenCalledWith('[Migrate Onyx] CheckForPreviousReportActionID Migration: previousReportActionID found. Migration complete');
+                    const connectionID = Onyx.connect({
+                        key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+                        waitForCollectionCallback: true,
+                        callback: (allReportActions) => {
+                            Onyx.disconnect(connectionID);
+                            const expectedReportAction1 = {};
+                            const expectedReportAction4 = {
+                                1: {
+                                    reportActionID: '1',
+                                    previousReportActionID: '10',
+                                },
+                                2: {
+                                    reportActionID: '2',
+                                    previousReportActionID: '23',
+                                },
+                            };
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]).toMatchObject(expectedReportAction1);
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]).toMatchObject(expectedReportAction4);
+                        },
+                    });
+                });
+        });
+
+        it('Should skip if no valid reportActions', () => {
+            const setQueries = {
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]: null,
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]: {},
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]: {},
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]: null,
+            };
+
+            // @ts-expect-error preset null values
```

Comment:
If it's possible to set `null` values maybe we can update `CollectionDataSet` type to include null:
```
type CollectionDataSet<TCollectionKey extends OnyxCollectionKey> = Record<`${TCollectionKey}${string}`, OnyxEntry<OnyxCollectionValuesMapping[TCollectionKey]>>;
```
After that you can type `setQueries` var using `CollectionDataSet` type and get rid of this comment. What do you think?

Reply:
there is a problem not only with a type. 
In this case, we need to specify the key in onyx, since  `toCollectionDataSet` we can set it only from the object data by passing the selector (in this case we will select from the )

I think the current approach is simpler for test purposes

Reply:
I mean if you update `CollectionDataSet` type as I mentioned above, you can then type it like

```
 const setQueries: CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS>  = {
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]: null,
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]: {},
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]: {},
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]: null,
            };
```
No need to use `toCollectionDataSet` in this case, and you will not have TS error. You can even create a separate type for `CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS>` as you did [here](https://github.com/Expensify/App/pull/37482/files#diff-947a4c1dd5fa2e942610b43b2397e682a2fea075ea054a04dc2c6b812b5f5717R7).

But it's up to you